nordicsemi/dfu/package.py

```diff
@@ -152,6 +155,9 @@ def __init__(self,
                                      filename=app_fw,
                                      init_packet_data=init_packet_vars)
 
+        if sd_req is not None:
+            init_packet_vars[PacketField.REQUIRED_SOFTDEVICES_ARRAY] = sd_req
```

Comment:
Sorry, I thought `init_packed_vars` was the final init packet, but I see now that `init_packet_vars` is just a temporary variable. The `init_packet_vars` is mutated with different softdevice requirements between each call to `__add_firmware_info()`. Then it makes sense.